Automattic/newspack-plugin
# Add setting for controlling MailChimp reader default status

### All Submissions:






### Changes proposed in this Pull Request:



Closes [an issue we've been encountering on a couple client sites](https://app.asana.com/0/1206477424334739/1206714953071858/f).

The original reasoning behind making the default status 'transactional' in MailChimp when people register on a site is correct at a strictly technical level because they technically haven't signed up to receive mailings from the site, but have merely registered on it. However **in practice** it is not the behavior publishers expect and it prevents them from doing e.g. a welcome series of emails when someone registers on the site.

This PR adds a new setting for controlling the default status of these readers in MailChimp. 

### How to test the changes in this Pull Request:

1. In the Engagement wizard, make sure you have this reader activity data setting set with an audience ID as well as some newsletters available in your Registration modal:

<img width="1065" alt="Screenshot 2024-03-07 at 2 26 52 PM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/971ddba7-de52-4702-88cc-988ecdbcb9f6">

2. You should now see a new setting **if an audience is set** for controlling the default status. Leave it in "Transactional/Non-Subscribed" for now.

<img width="883" alt="Screenshot 2024-03-15 at 10 14 00 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/2ed5e6db-b122-409d-a5b7-3c66fc694f52">

4. In an incognito window, register on the site but **do not** select any newsletters:

<img width="824" alt="Screenshot 2024-03-15 at 10 13 02 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/94a49854-9aec-4a9d-bfcd-28cbb1d5d95b">

5. In MailChimp, observe the reader is in "Non-Subscribed" status:

<img width="362" alt="Screenshot 2024-03-15 at 10 13 32 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/55a1e1f0-7e18-43bc-ba82-bdccf06f5a2f">

6. Set the setting to "Subscribed"

<img width="873" alt="Screenshot 2024-03-15 at 10 14 08 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/297c0b73-9e46-4cbd-bee2-811631a55982">

7. In an incognito window, register on the site again and **do not** select any newsletters:

<img width="801" alt="Screenshot 2024-03-15 at 10 19 02 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/5351bc0f-22d0-45fe-bdce-53483ec36cf7">

8. In MailChimp, observe the reader is in "Subscribed" status:

<img width="360" alt="Screenshot 2024-03-15 at 10 19 24 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/2584b1b9-1c5b-4f30-9c76-cd6f009cbb79">

9. Just to be safe, set the setting back to "Transactional/Non-Subscribed" and register on the site but **do** select a newsletter. In MailChimp the user should be in "Subscribed" status:

<img width="800" alt="Screenshot 2024-03-15 at 10 20 11 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/92c24512-611f-48a7-85f5-e6b23782e082">
<img width="349" alt="Screenshot 2024-03-15 at 10 20 35 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/b9024ff8-ca5d-44d2-967a-f919444f867c">


### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206740961392205
